### DISCUSSION TITLE
Interfacing Third Order IFCs with external thermal conductivity calculators

### DISCUSSION THREAD

**Discussion by dafolkner** [SSCHAcode/python-sscha#104] [Q&A] [answered]
I was wondering if the group has any examples of interfacing third order force constant results with external calculators like ShengBTE, Phono3py, or kALDO? I am attempting to implement SSCHA to calculate thermal properties, but there are no clear examples of how to utilize these higher order tensors for codes outside SSCHA.

**Comment by mesonepigreco** [comment, marked as answer]
Sorry for the  late reply,

Hi, it should be possible to save the third-order force constant matrix in phonopy format.

After initializing and centering Tensor3 object as explained in the tutorial http://sscha.eu/Tutorials/tutorial_spectral/
```

tensor3 = CC.ForceTensor.Tensor3(dyn.structure,
                                dyn.structure.generate_supercell(supercell),
                                supercell)

# Assign the tensor3 values
d3 = np.load("d3_realspace_sym.npy")*2.0
tensor3.SetupFromTensor(d3)

# Center and apply ASR
tensor3.Center()
tensor3.Apply_ASR()

# Print it
tensor3.WriteOnFile(fname="FC3",file_format='D3Q')
```

You replace the WriteOnFile file_format variable with "phonopy" and it should save it in a file readable by phono3py.
We will also soon release a thermal transport routine to compute thermal conductivity directly within the SSCHA package.
